os/mm/mm_heap/mm_sem.c

```diff
@@ -106,6 +106,11 @@ void mm_seminitialize(FAR struct mm_heap_s *heap)
 	 * private data sets.
 	 */
 
+	if (heap == NULL) {
```

Comment:
This function is used only from mm_initialize and that also gets heap addr as argu. I think we should check it in mm_initialize, not here.